chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2015/AttestedCriteriaCqmReviewer.java

```diff
@@ -23,62 +24,55 @@
 public class AttestedCriteriaCqmReviewer implements Reviewer {
     private ValidationUtils validationUtils;
     private ErrorMessageUtil msgUtil;
-    private CertificationCriterionDAO criteriaDao;
-
-    private static final String[] CRITERIA_REQUIRING_CQM = {
-            "170.315 (c)(1)", "170.315 (c)(2)", "170.315 (c)(3)", "170.315 (c)(4)"
-    };
+    private List<CertificationCriterion> cqmEligibleCriteria;
 
     @Autowired
-    public AttestedCriteriaCqmReviewer(ValidationUtils validationUtils, CertificationCriterionDAO criteriaDao,
-            ErrorMessageUtil msgUtil) {
+    public AttestedCriteriaCqmReviewer(ValidationUtils validationUtils,
+            CertificationCriterionService criteriaService,
+            ErrorMessageUtil msgUtil,
+            @Value("${cqmEligibleCriteria}") String cqmCriteriaIdList) {
         this.validationUtils = validationUtils;
-        this.criteriaDao = criteriaDao;
         this.msgUtil = msgUtil;
+
+        cqmEligibleCriteria = Arrays.asList(cqmCriteriaIdList.split(",")).stream()
+                .map(id -> criteriaService.get(Long.parseLong(id)))
+                .collect(Collectors.toList());
     }
 
     @Override
     public void review(CertifiedProductSearchDetails listing) {
-        List<CertificationCriterion> cqmEligibleCritera = new ArrayList<CertificationCriterion>();
-        for (String criterionNumber : CRITERIA_REQUIRING_CQM) {
-            List<CertificationCriterionDTO> criteriaDtos = criteriaDao.getAllByNumber(criterionNumber);
-            cqmEligibleCritera.addAll(criteriaDtos.stream()
-                    .map(dto -> new CertificationCriterion(dto))
-                    .collect(Collectors.<CertificationCriterion>toList()));
-        }
-
         //any attested criteria that is eligible for CQM must have a CQM that references it
         List<CertificationCriterion> attestedCriteria = validationUtils.getAttestedCriteria(listing);
-        for (CertificationCriterion criterion : attestedCriteria) {
-            if (isCriteriaEligibleForCqm(criterion, cqmEligibleCritera)) {
-                //is there a cqm with this criterion?
-                if (!isCriterionIncludedInCqms(criterion, listing.getCqmResults())) {
-                    listing.getErrorMessages().add(msgUtil.getMessage("listing.criteria.missingCqmForCriteria",
-                            Util.formatCriteriaNumber(criterion)));
-                }
-            }
-        }
+        attestedCriteria.stream()
+            .filter(attestedCriterion -> isCriterionEligibleForCqm(attestedCriterion))
+            .filter(attestedCqmEligibleCriterion -> !isCriterionInAnyCqm(attestedCqmEligibleCriterion, listing.getCqmResults()))
```

Comment:
Discussion point and not necessarily a code change required - This requires the list to be gone through twice.  By combining these lines into a single `filter` using the `&&` it will be technically be more efficient.  But I do find your style to be more readable.